lib/internal/util.js

```diff
@@ -187,3 +187,18 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+exports.getConstructorOf = function getConstructorOf(obj) {
+  while (obj) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, 'constructor');
```

Comment:
That would be a breaking change for a class that defines `constructor` as accessors.